File: app/src/main/java/com/v2/desafionubank/di/component/FragmentComponent.java
package com.v2.desafionubank.di.component;

import com.v2.desafionubank.di.PerFragment;
import com.v2.desafionubank.di.module.FragmentModule;
import com.v2.desafionubank.ui.BaseFragment;

import dagger.Component;

/**
 * Created by CaioSChristino on 20/02/2018.
 */

@PerFragment
@Component(dependencies = ApplicationComponent.class, modules = FragmentModule.class)
public interface FragmentComponent {
    void inject(BaseFragment mFragment);
}

File: app/src/main/java/com/v2/desafionubank/di/component/FragmentComponent.kt
package com.v2.desafionubank.di.component

import com.v2.desafionubank.di.PerFragment
import com.v2.desafionubank.di.module.FragmentModule
import com.v2.desafionubank.ui.BaseFragment
import dagger.Component

/**
 * Created by csanchez on 19/04/2018.
 */

@PerFragment
@Component(dependencies = arrayOf(ApplicationComponent::class), modules = arrayOf(FragmentModule::class))
interface FragmentComponent {
    fun inject(mFragment: BaseFragment)
}